scrapers_next/nd/people.py

```diff
@@ -1,118 +1,160 @@
+        p.add_source(first_list_page, "first page of members list")
 
         return p
 
 
 class LegList(HtmlListPage):
-    source = URL(
-        "https://www.legis.nd.gov/assembly/67-2021/members/members-by-district"
+    session = "67-2021"
+    source = f"https://www.ndlegis.gov/assembly/{session}/regular/members"
+    selector = XPath("//div[@class='member-wrapper']")
+    next_page_selector = XPath(
+        "//nav[@class='pager']/ul/li[@class='page-item pager__item--next']/a/@href"
     )
-    selector = CSS("div.view-content > div", min_items=142)
+
+    def get_next_source(self):
+
+        try:
+            next_urls = self.next_page_selector.match(self.root)
+        except SelectorError:
+            return
+
+        if len(set(next_urls)) > 1:
+            raise PaginationError(
+                f"get_next_source returned multiple links: {next_urls}"
+            )
+
+        return next_urls[0]
 
     def process_item(self, item):
-        name = CSS("div.name").match_one(item).text_content().strip()
-        name = re.search(r"(Senator|Representative)\s(.+)", name).groups()[1]
-        # Luke Simons was expelled on 3/4/21
-        if name == "Luke Simons":
-            self.skip()
-
-        chamber = CSS("div.chamber").match_one(item).text_content().strip()
-        if chamber == "Senate":
-            chamber = "upper"
-        elif chamber == "House":
-            chamber = "lower"
+        img_elem = item.find_class("image-style-member-list-photo")[0]
+        img_src = img_elem.get("src")
 
-        for previous_tag in item.itersiblings(preceding=True):
-            if previous_tag.get("class") == "title":
-                district = previous_tag.text_content().strip()
-                district = re.search(r"District\s(.+)", district).groups()[0]
-                break
+        name_wrapper = item.find_class("strong member-name")[0]
+        name_tag = name_wrapper.getchildren()[0]
+        member_name = name_tag.text_content().strip()
 
-        party = CSS("div.party").match_one(item).text_content().strip()
-        if party == "Democrat":
-            party = "Democratic"
+        chamber_div = item.find_class("strong member-chamber")[0]
+        chamber = chamber_div.text_content().strip()
+        if chamber == "Senator":
+            chamber = "upper"
+        else:
+            chamber = "lower"
 
-        p = ScrapePerson(
-            name=name,
-            state="nd",
+        status = item.find_class("member-status")
+        start_date = ""
+        if status:
+            status_text = status[0].text_content().strip()
+            if "Expelled" in status_text:
+                self.skip()
+            elif "Active" in status_text:
+                start_date = status_text.split()[-1].strip()
+
+        district_div = item.find_class("district")[0].text_content().strip()
+        dist_match = re.search(r"District\s+\S+", district_div)
```

Comment:
Should try to capture the district number in the search using something like [this](https://github.com/openstates/openstates-scrapers/blob/f97fe365665733a343e1addbf4f19ef25e5417bf/scrapers_next/tx/people.py#L86) so you don't need the next few lines

Reply:
I implemented [this one-line solution](https://github.com/openstates/openstates-scrapers/pull/4162/commits/ed81cdcef99f666fd2d8464c036fe5263b687dee#diff-b4eb1b6b6189a33b171b390670cb2f7c25db0b25acd7e596c8c5afb685d778a6R136) in the most updated commit (which is on nd_events branch).